great_expectations/core/expectation_suite.py

```diff
@@ -133,6 +134,45 @@ def __init__(  # noqa: PLR0913
     def name(self) -> str:
         return self.expectation_suite_name
 
+    @property
+    def expectations(self) -> list[Expectation]:
```

Comment:
Should uses be thinking of this list as ordered or as a set?

If a set, why does re-adding cause value error?

Reply:
this change was a part of a previous PR, but your right, no reason for this to be ordered.

Reply:
I've updated the docstring to be clear that `ExpectationSuite` is set-like. I also debated renaming `delete` to `remove` to match the set API, but decided consistency with other GX objects is probably clearer.